### ISSUE TITLE
Roboclaw

### ISSUE THREAD

**Issue by Rafcin** [avirtuos/ros_hercules#4]
Is it possible to swap out the serial communication for the wheels with a Roboclaw motor controller node that instead uses values such as:

0 Shuts Down Channel 1 and 2
1 Channel 1 - Full Reverse
64 Channel 1 - Stop
127 Channel 1 - Full Forward
128 Channel 2 - Full Reverse
192 Channel 2 - Stop
255 Channel 2 - Full Forward

I just think it would be much more efficient than wires to a jetson becuase instead it works over usb if you use a Roboclaw 2x30A(v4).

**Comment by avirtuos** [comment]
I don't have wires to the jetsob, there is an Arduino that controls these
and there is an ros_serial node that handles the handoff.

**Comment by avirtuos** [comment]
I should also mention that the motor control I'm using is in the parts list and it support serial directly as well. I used the Arduino to consolidate several serial interfaces.

**Comment by Rafcin** [comment]
Ah, interesting, ok. I'm going to sit-down and try to see if I can get it to work with the RoboClaw node from Aion-R1, if so I'll post what here what I did.

**Comment by avirtuos** [comment]
Let me know. Happy to help if needed.

**Comment by avirtuos** [comment]
Also, I'm rewriting this project so that every component is an ros_node using a raspberry pi zero w. It made the robot more modular and easier to reuse parts in different projects.

**Comment by Rafcin** [comment]
Interesting, I can try to help the best I can, i'm actually working on a rescue robot that uses a Jetson Tx2. I'll do a fork of this project and see if I can provide anything new.

**Comment by Rafcin** [comment]
ROS is definitely weird to work with but I made a few launch files for mapping, and sensors and then a master launch file with all the nodes and an RVIZ menu. Hector mapping is odd to use :O

**Comment by Rafcin** [comment]
Hey sorry to bother you but I had a question about your hercules.launch file. So I've look at how your Rover works and done some very similar things, but my goal is to make it work outside and build a map, the issue is instead of using the lidar to map because there are no walls I'm using the depth scan converter and passing it to hector node, but when visualising it, Zed Odom works fine, and the laser scan TF decides that any slight movement pisses it off and it drifts across the map, did you encounter something like this ever, is it possible the depth to laser node isn't proper for this?

**Comment by avirtuos** [comment]
@Rafcin yes, I did and still do face this issue. What I've seen is that there are really only two solutions...

1. Get more frequent data (laser scan or pointCloud2LaserScan in the case of Zed). This  increases the probability that the SLAM module will be able to expand the map and locate position... Basically more frequent data means the difference between each update is less. 

2. Go slower... If your Robert alows down it has the same effect... Less difference between map updates.

**Comment by avirtuos** [comment]
Another option is to use something other than Hector SLAM which might be a bit better... But those toher algos also need more CPU power.

**Comment by Rafcin** [comment]
I see, well a Jetson is strong, maybe RTAB Map

**Comment by AmirsamanAhmadi** [comment]
@avirtuos Hi Hope you are doing fine.  earlier you mentioned  

> Also, I'm rewriting this project so that every component is an ros_node using a raspberry pi zero w. It made the robot more modular and easier to reuse parts in different projects.

May I know if you finish it ?
